Source/WebKit/UIProcess/ios/WKContentViewInteraction.mm

```diff
@@ -10749,16 +10749,20 @@ - (void)dragInteraction:(UIDragInteraction *)interaction item:(UIDragItem *)item
 {
     _isAnimatingDragCancel = YES;
     RELEASE_LOG(DragAndDrop, "Drag interaction willAnimateCancelWithAnimator");
-    auto previewView = _dragDropInteractionState.takePreviewViewForDragCancel();
-    [previewView setAlpha:0];
-    [animator addCompletion:[protectedSelf = retainPtr(self), previewView = WTFMove(previewView), page = _page] (UIViewAnimatingPosition finalPosition) {
+    auto previewViews = _dragDropInteractionState.takePreviewViewsForDragCancel();
+    for (auto& previewView : previewViews)
+        [previewView setAlpha:0];
+
+    [animator addCompletion:[protectedSelf = retainPtr(self), previewViews = WTFMove(previewViews), page = _page] (UIViewAnimatingPosition finalPosition) {
         RELEASE_LOG(DragAndDrop, "Drag interaction willAnimateCancelWithAnimator (animation completion block fired)");
-        [previewView setAlpha:1];
-        page->dragCancelled();
-        page->callAfterNextPresentationUpdate([previewView = WTFMove(previewView), protectedSelf = WTFMove(protectedSelf)] {
-            [previewView removeFromSuperview];
-            protectedSelf->_isAnimatingDragCancel = NO;
-        });
+        for (auto& previewView : previewViews) {
+            [previewView setAlpha:1];
+            page->dragCancelled();
```

Comment:
Nit - from code inspection, it seems safe (but unnecessary) to repeatedly call `dragCancelled()` for each `previewView`.

If possible, we should:

1. Reveal all of the preview views (`-setAlpha:1`)
2. Call `dragCancelled()` once, and then `callAfterNextPresentationUpdate`.
3. In the presentation update callback, remove all of the preview views (`-removeFromSuperview`).